pkg/build/util/generate.go

```diff
@@ -30,3 +31,44 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 		},
 	}
 }
+
+// SubstituteImageReferences replaces references to an image with a new value
+func SubstituteImageReferences(build *api.Build, oldImage string, newImage string) {
+	if build.Parameters.Strategy.Type==api.DockerBuildStrategyType && 
+	   build.Parameters.Strategy.DockerStrategy!=nil{
+		if build.Parameters.Strategy.DockerStrategy.BaseImage==oldImage {
+			build.Parameters.Strategy.DockerStrategy.BaseImage=newImage
+		}		
+	} else if build.Parameters.Strategy.Type==api.STIBuildStrategyType &&
+			  build.Parameters.Strategy.STIStrategy!=nil {
+		if build.Parameters.Strategy.STIStrategy.Image==oldImage {
+			build.Parameters.Strategy.STIStrategy.Image=newImage
```

Comment:
In fact, looks like others are missing it to - be sure to hook gofmt up to your editor.